### PR TITLE
Secure credentials

### PR DESCRIPTION
**Pull request by bleibdirtroy**
- [x] function to store key value pair
- [x] function to read value from key
- [x] function to delete all keys
- [x] function to update existing value of key
- [x] function to check if key exists
- [x] skip login screen if host, username and password are stored
- [x] delete host, username and password if logged out
- [x] exceptions and tests

### CODE REVIEW

**Review by johnns-s — CHANGES_REQUESTED** [review]

File: lib/services/key_value_storage/secure_storage_provider.dart [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      await _storage.write(
+        key: key,
+        value: value,
+        aOptions: _getAndroidOptions(),
+        iOptions: _getIOSOptions(),
+      );
+    } else {
+      throw KeyAlreadyExistsInStorage();
+    }
+  }
```
Comment (johnns-s) [inline lib/services/key_value_storage/secure_storage_provider.dart:32]: When I try to login by just using a URL the app crashes with an KeyAlreadyExistsInStorage-Exception

**Comment by johnns-s** [comment]
Why are the fields Username and Password allowed to be empty. I think all 3 fields should not be allowed to be empty.

**Comment by bleibdirtroy** [comment]
Because a public photoprism instance does not need username and password.

File: lib/services/key_value_storage/secure_storage_provider.dart (hunk 1 above)
Comment (johnns-s) [reply lib/services/key_value_storage/secure_storage_provider.dart:32]: Can not login at all.

**Review by johnns-s — CHANGES_REQUESTED** [review]

**Review by johnns-s — APPROVED** [review]

**Pull request merged**